src/community/label_propagation.jl

```diff
@@ -0,0 +1,99 @@
+"""
+Community detection using the label propagation algorithm (see [Raghavan et al.](http://arxiv.org/abs/0709.2938)).
+`g`: imput Graph
+`maxiter`: optional, maximum iterations
+return : array containing vertex assignments, the range of the output as 1:countdistinct(membership)
+"""
+function label_propagation(g::SimpleGraph; maxiter=1000)
+    n = nv(g)
+    membership = collect(1:n)
+    active_nodes = Set(vertices(g))
+    nc = NeighComm(collect(1:n), fill(-1,n), 1)
+    i = 0
+    while !isempty(active_nodes) && i < maxiter
+    	i += 1
+        random_order = shuffle(collect(active_nodes))
```

Comment:
This collect allocates a length(active_nodes) array. but then you do an inplace shuffle to avoid an allocation. Is that what you wanted?